test/_vagrant/vagrant_test.go

```diff
@@ -0,0 +1,204 @@
+package vagrant_test
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/gianarb/vagrant-go"
+	"github.com/tinkerbell/tink/client"
+	"github.com/tinkerbell/tink/pkg"
+	"github.com/tinkerbell/tink/protos/hardware"
+	"github.com/tinkerbell/tink/protos/template"
+	"github.com/tinkerbell/tink/protos/workflow"
+)
+
+func TestVagrantSetupGuide(t *testing.T) {
+	ctx := context.Background()
+
+	machine, err := vagrant.Up(ctx,
+		vagrant.WithLogger(t.Logf),
+		vagrant.WithMachineName("provisioner"),
+		vagrant.WithWorkdir("../../deploy/vagrant"),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer func() {
+		err := machine.Destroy(ctx)
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	_, err = machine.Exec(ctx, "cd /vagrant/deploy && source ../envrc && docker-compose up -d")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = machine.Exec(ctx, "docker pull hello-world")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = machine.Exec(ctx, "docker tag hello-world 192.168.1.1/hello-world")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = machine.Exec(ctx, "docker push 192.168.1.1/hello-world")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for ii := 0; ii < 5; ii++ {
+		resp, err := http.Get("http://localhost:42114/healthz")
+		if err != nil || resp.StatusCode != http.StatusOK {
+			if err != nil {
+				t.Logf("err tinkerbell healthcheck... retrying: %s", err)
+			} else {
+				t.Logf("err tinkerbell healthcheck... expected status code 200 got %d retrying", resp.StatusCode)
+			}
+			time.Sleep(10 * time.Second)
+		}
+	}
+
+	t.Log("Tinkerbell is up and running")
+
+	os.Setenv("TINKERBELL_CERT_URL", "http://127.0.0.1:42114/cert")
+	os.Setenv("TINKERBELL_GRPC_AUTHORITY", "127.0.0.1:42113")
+	client.Setup()
+	_, err = client.HardwareClient.All(ctx, &hardware.Empty{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = registerHardware(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	templateID, err := registerTemplate(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Logf("templateID: %s", templateID)
+
+	workflowID, err := createWorkflow(ctx, templateID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Logf("WorkflowID: %s", workflowID)
+
+	os.Setenv("VAGRANT_WORKER_GUI", "false")
+	worker, err := vagrant.Up(ctx,
+		vagrant.WithLogger(t.Logf),
+		vagrant.WithMachineName("worker"),
+		vagrant.WithWorkdir("../../deploy/vagrant"),
+		vagrant.RunAsync(),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer func() {
+		err := worker.Destroy(ctx)
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	for iii := 0; iii < 30; iii++ {
```

Comment:
Just curious, why a `iii` and not a `i`?

Reply:
I tend to use multi `i` for the index because sometimes I use `i` for other things :D so in order to avoid confusions I just have this method stuck in my head :D